server/mqtt.go

```diff
@@ -1523,51 +1528,112 @@ func (jsa *mqttJSA) prefixDomain(subject string) string {
 	return subject
 }
 
-func (jsa *mqttJSA) newRequestEx(kind, subject, cidHash string, hdr int, msg []byte, timeout time.Duration) (interface{}, error) {
-	var sb strings.Builder
-	// Either we use nuid.Next() which uses a global lock, or our own nuid object, but
-	// then it needs to be "write" protected. This approach will reduce across account
-	// contention since we won't use the global nuid's lock.
-	jsa.mu.Lock()
-	uid := jsa.nuid.Next()
-	sb.WriteString(jsa.rplyr)
-	jsa.mu.Unlock()
-
-	sb.WriteString(kind)
-	sb.WriteByte(btsep)
-	if cidHash != _EMPTY_ {
-		sb.WriteString(cidHash)
-		sb.WriteByte(btsep)
+func (jsa *mqttJSA) newRequestEx(kind, subject, cidHash string, hdr int, msg []byte, timeout time.Duration) (any, error) {
+	responses, err := jsa.newRequestExMulti(kind, subject, cidHash, []int{hdr}, [][]byte{msg}, timeout)
+	if err != nil {
+		return nil, err
+	}
+	if len(responses) != 1 {
+		return nil, fmt.Errorf("unreachable: invalid number of responses (%d)", len(responses))
+	}
+	v := responses[0].value
```

Comment:
Again, no, if newRequestExMulti() returns an error, we would have already returned at line 1533. For instance, here is how `loadNextMsgFor()` looks like:
```
lmri, err := jsa.newRequest(mqttJSAMsgLoad, fmt.Sprintf(JSApiMsgGetT, streamName), 0, req)
if err != nil {
    return nil, err
}
lmr := lmri.(*JSApiMsgGetResponse)
return lmr.Message, lmr.ToError()
```
And before your changes, `newRequestEx()` would have returned nil and an error on timeout. With your changes, you return the responses and the error, and if there is a timeout, the error will have been handled line 1533 as I said. If the error is part of the response object, it is up to the caller to use it. See above, we cast the response to `JSApiMsgGetResponse` and then return the message part and the error part (that may be nil).

Again, I fail to see why returning `responses[0].value, nil` here is not enough, and consistent with previous behaviour.

Reply:
You're right, I over-complicated this. Fixing.